core/modules/node/node.pages.inc

```diff
@@ -717,7 +717,18 @@ function node_preview_banner_form_node_submit(array $form, array &$form_state) {
     }
   }
   else {
-    $destination = node_access('view', $node) ? 'node/' . $node->nid : '<front>';
+    // Determine whether to redirect user to full page display.
+    $destination = '<front>';
+    if (node_access('view', $node)) {
```

Comment:
I could be wrong, but this seems that it would be better as an `elseif`, with the `$destination = '<front>';` then left alone as a last `if`. Either that, or we should be using more "wordy" comments here to explain what's going on.

Reply:
There's a couple good reasons *not* to do it that way:

1. Regarding hidden paths: we only care about when they have access to view the node (e.g. they have "view own published content" permission") and then need to check if they have the hidden path restriction and permission.
2. We set `$destination = <front>` at the beginning since `$destination` needs to be defined regardless of the if statements (we don't want a situation where `$destination` ends up being undefined). We also know if that the following conditions don't hold that the user should just be redirected to the front page. This logic still matches that of the old logic but with the added check.

Regardless of the above, the logic should match that in `node_form_submit()` which it does currently. If we want to change it around we can do it in a new issue I imagine.

Reply:
Fair enough I guess.

Reply:
Thanks @klonos. Can you mark the issue as RTBC if you agree with the approach? And not sure if you want to create a follow-up or not.